src/inc/refresh-token.ts

```diff
@@ -1,37 +1,40 @@
 import jwtDecode, { InvalidTokenError } from 'jwt-decode'
 import { addTokenPrefix } from '../utils'
-import type { Auth } from '../'
+import type { Scheme } from '../'
+import Storage from '../core/storage'
 import TokenStatus from './token-status'
 
 export default class RefreshToken {
-  public $auth: Auth
+  public scheme: Scheme
+  public $storage: Storage
 
-  constructor (auth: Auth) {
-    this.$auth = auth
+  constructor (scheme: Scheme, storage: Storage) {
+    this.scheme = scheme
```

Comment:
However we call scheme instance strategy, i like your idea for simplification. Maybe can always call it scheme :+1: